documentation/src/main/asciidoc/ch11.asciidoc

```diff
@@ -694,17 +694,25 @@ mechanism to register additional constraint definitions. All you have to do is t
 _javax.validation.ConstraintValidator_ to _META-INF/services_. In this service file you list the
 fully qualified classnames of your constraint validator classes (one per line). Hibernate Validator
 will automatically infer the constraint types they apply to.
-See <<example-using-service-file-for-constraint-definitions>> for an example.
+See <<example-using-service-file-for-constraint-definitions,Constraint definition via service file>>
+for an example.
 
 [[example-using-service-file-for-constraint-definitions]]
-.Using _META-INF/services/javax.validation.ConstraintValidator_ for constraint definitions
+._META-INF/services/javax.validation.ConstraintValidator_
+
 ====
 [source]
 ----
+# Assuming a custom constraint annotation @org.mycompany.CheckCase
 org.mycompany.CheckCaseValidator
 ----
 ====
 
+To contribute default messages for your custom constraints, place a _ValidationMessages.properties_
```

Comment:
> That's what I meant to address when saying "which then is 'merged' (e.g. similar to service loader files)". I.e. we'd consider all the entries from all the ContributorValidationMessages bundles. 

Ahh. You realize that this will require some doing? `ResourceBundle.getBundle` internally just returns any resource bundle of the requested name (uses `Classloader#getResource internally`). There is afaict no way to out of the box get all bundles with the same name. There is a `ResourceBundle.Control` which one apparently can use to customize some of the loading behaviour, but there is of course a risk of getting it wrong. 

The idea is nice, but I am afraid, it will take too much time. I can experiment a bit.